action.yml

```diff
@@ -26,9 +26,13 @@ runs:
     - name: Restore snapshot
       run: ${{ inputs.snaplet-restore-command }}
       shell: bash
+      env:
+        SNAPLET_TARGET_DATABASE_URL: ${{ inputs.snaplet-database-url }}
 
     - name: Commit and push Docker image
       run: |
-        docker commit ${{ inputs.docker-container-name }} ${{ inputs.docker-container-tag }}
+        CONTAINER_PORT=$(echo "${{ inputs.snaplet-database-url }}" | sed 's/.*:\([0-9]*\)\/.*/\1/;s/^$/5432/')
```

Comment:
**question**

As I like the idea to infer container name from the `database-url` reducing the number of arguments. I also know that database-url can be much more tricky to parse than this sed command line.

It will probably work in 90% of the case so I'm all for keeping this. I would just allow an optional input to override this if the user want to for some reason.

Something like:

```
      CONTAINER_ID=$(docker ps -f publish=$CONTAINER_PORT --format {{.ID}})
      DOCKER_CONTAINER_NAME=${{ inputs.docker-container-name }}
      CONTAINER_NAME=${DOCKER_CONTAINER_NAME:-$CONTAINER_ID}
      docker commit $CONTAINER_NAME ${{ inputs.docker-container-tag }}
```

Would allow the user to override the computed value if he needs to, but keep the per default behaviour easy to use.

What do you think ?

Reply:
I think it is a sensible change that provides additional flexibility.

Reply:
After giving it another thought, I have decided to revert back to using only service container name. It's simpler and hence more error prone. I have updated the action accordingly.

Reply:
Hey ! Nice, after thinking more about it, I also think this is probably more robust and simple to understand. Thank's for that.